### PR TITLE
[Refactor] Verify RLE on MPII

### PR DESCRIPTION
**Pull request by Tau-J**
## Motivation
support pckh on mpii dataset

## Modification

modified `base_coco_style_dataset.py` to add headbox_file
fix bug on mpii dataset

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmpose/datasets/datasets/body/mpii_dataset.py [hunk 1]
```diff
@@ -115,16 +131,25 @@ def _load_annotations(self) -> List[dict]:
                 keypoints = np.zeros((1, num_keypoints, 2), dtype=np.float32)
```
Comment (Tau-J) [inline mmpose/datasets/datasets/body/mpii_dataset.py:131]: Now I'm curious about why we use dummy data as keypoints on test_mode, which result in a incorrect PCKh computation.

Comment (liqikai9) [reply mmpose/datasets/datasets/body/mpii_dataset.py:131]: I suppose this is to be aligned with the previous version: https://github.com/open-mmlab/mmpose/blob/master/mmpose/datasets/datasets/top_down/topdown_mpii_dataset.py#L113.

Intuitively, the ground truth keypoints positions are unknown and also unnecessary when `test_mode` is `True`.

Comment (Tau-J) [reply mmpose/datasets/datasets/body/mpii_dataset.py:131]: But in previoius version, the metric `PCKh` is calculated in `evaluation()` in mpii_dataset, which loads keypoints and head box from the `.mat` file individually. But right now, we need to load the head boxes infomation and pass them in gt_instances. Therefore, maybe we should change the codes here.

File: mmpose/datasets/datasets/base/base_coco_style_dataset.py [hunk 2]
```diff
@@ -63,6 +63,7 @@ class BaseCocoStyleDataset(BaseDataset):
     def __init__(self,
                  ann_file: str = '',
                  bbox_file: Optional[str] = None,
+                 headbox_file: Optional[str] = None,
```
Comment (ly015) [inline mmpose/datasets/datasets/base/base_coco_style_dataset.py:66]: If so far `headbox_file` is only for the mpii dataset, I suggest only adding it in ``MpiiDataset``

**Comment by ly015** [comment]
Maybe use `ori_keypoints`? MMCV uses `ori_shape` so we can use the same convention.
https://github.com/open-mmlab/mmcv/blob/dev-2.x/mmcv/transforms/loading.py#L81

File: configs/tests/res50_mpii_256x256.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
+_base_ = [
```
Comment (ly015) [inline configs/tests/res50_mpii_256x256.py:1]: Please move the verified configs to proper locations like `configs/body2d_keypoints/topdown_regression/mpii/'.

**Review by ly015 — APPROVED** [review]

**Pull request merged**